internal/output/output_result.go

```diff
@@ -0,0 +1,538 @@
+package output
+
+import (
+	"cmp"
+	"encoding/json"
+	"io"
+	"slices"
+	"strings"
+
+	"github.com/google/osv-scanner/internal/identifiers"
+	"github.com/google/osv-scanner/internal/semantic"
+	"github.com/google/osv-scanner/internal/utility/severity"
+	"github.com/google/osv-scanner/pkg/models"
+)
+
+// OutputResult represents the vulnerability scanning results for output report.
+type OutputResult struct { //nolint:revive
```

Comment:
Why is there nolint for all of these types, can we resolve the lint instead?

Guessing it's caused by stuttering? Do you know where outside this package requires these types, if it is not required, let's make all of these private.

Reply:
It's because of the naming.
`exported: type name will be used as output.OutputResult by other packages, and that stutters; consider calling this Result (revive)`
Any objection if I rename it from `OutputResult` to `Result`?

Reply:
Changed name to "Result"